src/map.cpp

```diff
@@ -3622,211 +3622,92 @@ void map::crush( const tripoint &p )
+
+        // if you are aiming at this tile, you can never miss
+        if( hit_items || x_in_y( shoot.chance_to_hit, 100 ) ) {
+            if( laser ) {
+                dam -= rng( shoot.reduce_dmg_min_laser, shoot.reduce_dmg_max_laser );
+            } else {
+                dam -= rng( shoot.reduce_dmg_min, shoot.reduce_dmg_max );
+            }
+            // lasers can't destroy some types of furn/ter you can shoot through
+            if( !laser || !shoot.no_laser_destroy ) {
+                // important to use initial damage, energy from reduction has gone into the furn/ter
+                const int x = int( initial_damage ) - shoot.destroy_dmg_min;
+                const int y = shoot.destroy_dmg_max - shoot.destroy_dmg_min;
+                if( x_in_y( x, y ) ) {
+                    // don't need to duplicate all the destruction logic here
+                    bash_params bsh{ 0, false, true, false, 0.0, false, false, false, false };
+                    bash_ter_furn( p, bsh );
+                    destroyed = true;
                 }
             }
-        }
-    } else if( terrain == t_ballistic_glass || terrain == t_reinforced_glass ||
-               terrain == t_reinforced_door_glass_c
-               || terrain == t_reinforced_glass_shutter || terrain ==  t_reinforced_glass_shutter_open ) {
-        // reinforced glass stops most bullets
-        // laser beams are attenuated
-        if( ammo_effects.count( "LASER" ) ) {
-            dam -= rng( 0, 8 );
-        } else {
-            //Greatly weakens power of bullets
-            dam -= 40;
             if( dam <= 0 && get_player_view().sees( p ) ) {
-                if( terrain == t_reinforced_door_glass_c ) {
-                    add_msg( _( "The shot is stopped by the reinforced glass door!" ) );
-                } else {
-                    add_msg( _( "The shot is stopped by the reinforced glass wall!" ) );
-                }
-            } else if( dam >= 40 ) {
-                //high powered bullets penetrate the glass, but only extremely strong
-                // ones (80 before reduction) actually destroy the glass itself.
-                break_glass( p, 16 );
-                ter_set( p, t_floor );
+                // TODO: add exclamation mark once out of string freeze
+                add_msg( _( "The shot is stopped by the %s" ), data.name() );
+            }
+            // only very flammable furn/ter can be set alight with incendiary rounds
+            if( incendiary && data.has_flag( TFLAG_FLAMMABLE_ASH ) ) {
+                add_field( p, fd_fire, 1 );
+            }
+            // bash_ter_furn already triggers the alarm
+            // TODO: fix alarm event weirdness (not just here, also in bash, hack, etc)
+            if( !destroyed && data.has_flag( "ALARMED" ) &&
+                !get_timed_events().queued( timed_event_type::WANTED ) ) {
+                sounds::sound( p, 40, sounds::sound_t::alarm, _( "an alarm go off!" ),
```

Comment:
```suggestion
                sounds::sound( p, 40, sounds::sound_t::alarm, _( "an alarm sound!" ),
```
As well as grammar, I think this will remove the need for String Freeze.

Reply:
That was changed to match https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/map.cpp#L3213
I was under the impression that strings could be changed to something that already exists elsewhere.

Doing a quick search through the rest of the code base, it seems "alarm sound" is used in other places as well, but I think at least bashing and shooting should give the same message.

Reply:
Sorry; missed that other use, and you are correct that it's OK if the string is already elsewhere.

But is there some reason not to use "alarm sound" both in L3213 and in L3686? "an alarm go off" is neither grammatically correct nor - more importantly to people who don't get a near-literal headache looking at it - a description of the sound, which from looking at the other uses of `sounds::sound` is what the argument in question is supposed to be.

Reply:
oh, you're right, "alarm go off" is ONLY used in the one place, "alarm sound" is everywhere else. I just chose "alarm go off" since it's the one I'm used to seeing in game, since it occurs when bashing, and it sounds more natural to me. "alarm sound" sounds odd to me. It seems to apply that the survivor is thinking that the sound could come from something other than an alarm, otherwise the message would just be "you hear an alarm!" or alarm go off, etc. 

Although I suppose the sound can also come from a mi-go.

Reply:
"you hear an alarm sound" can be interpreted a couple of ways - you hear an alarm making a sound, or you hear a sound that is characteristic of an alarm.

> Although I suppose the sound can also come from a mi-go.

Currently not in speech.json, but that does make sense.